components/suggestion/command_provider/app_command_parser/app_command_parser.ts

```diff
@@ -0,0 +1,1256 @@
+                this.incomplete = '';
+                this.incompleteStart = this.i;
+                switch (c) {
+                case '"': {
+                    this.state = ParseState.QuotedValue;
+                    this.i++;
+                    break;
+                }
+                case '`': {
+                    this.state = ParseState.TickValue;
+                    this.i++;
+                    break;
+                }
+                case ' ':
+                case '\t':
+                    return this.asError(this.intl.formatMessage({
+                        id: 'apps.error.parser.unexpected_whitespace',
+                        defaultMessage: 'Unreachable: Unexpected whitespace.',
+                    }));
+                default: {
+                    this.state = ParseState.NonspaceValue;
+                    break;
+                }
+                }
+                break;
+            }
+
+            case ParseState.NonspaceValue: {
+                switch (c) {
+                case '':
+                case ' ':
+                case '\t': {
+                    this.state = ParseState.EndValue;
+                    break;
+                }
+                default: {
+                    this.incomplete += c;
+                    this.i++;
+                    break;
+                }
+                }
+                break;
+            }
+
+            case ParseState.QuotedValue: {
+                switch (c) {
+                case '': {
+                    if (!autocompleteMode) {
+                        return this.asError(this.intl.formatMessage({
+                            id: 'apps.error.parser.missing_quote',
+                            defaultMessage: 'Matching double quote expected before end of input.',
+                        }));
+                    }
+                    return this;
+                }
+                case '"': {
+                    if (this.incompleteStart === this.i - 1) {
+                        return this.asError(this.intl.formatMessage({
+                            id: 'apps.error.parser.empty_value',
+                            defaultMessage: 'empty values are not allowed',
+                        }));
+                    }
+                    this.i++;
+                    this.state = ParseState.EndQuotedValue;
+                    break;
+                }
+                case '\\': {
+                    escaped = true;
+                    this.i++;
+                    break;
+                }
+                default: {
+                    this.incomplete += c;
+                    this.i++;
+                    if (escaped) {
+                        //TODO: handle \n, \t, other escaped chars
+                        escaped = false;
+                    }
+                    break;
+                }
+                }
+                break;
+            }
+
+            case ParseState.TickValue: {
```

Comment:
Is there any difference between using quotes and backticks for a flag?

Reply:
Minor - like in Go, quoted strings have special meaning for `\`; backticks pass the string as is, do not allow for a backtick in the middle.